services/wallet/router.go

```diff
@@ -692,31 +738,40 @@ func (r *Router) suggestedRoutes(
 						gasLimit = sendType.EstimateGas(r.s, network, addrFrom, tokenID)
 					}
 
-					requiredNativeBalance := new(big.Int).Mul(gweiToWei(maxFees), big.NewInt(int64(gasLimit)))
-					// Removed the required fees from maxAMount in case of native token tx
-					if token.IsNative() {
-						maxAmountIn = (*hexutil.Big)(new(big.Int).Sub(maxAmountIn.ToInt(), requiredNativeBalance))
+					approvalRequired, approvalAmountRequired, approvalGasLimit, l1ApprovalFee, approvalContractAddress, err := r.requireApproval(ctx, sendType, bridge, addrFrom, network, token, amountIn)
+					if err != nil {
+						continue
 					}
+
+					l1GasFeeWei, _ := bridge.GetL1Fee(ctx, network, addrTo, addrFrom, token, amountIn)
+					l1GasFeeWei += l1ApprovalFee
+					gasFees.L1GasFee = weiToGwei(big.NewInt(int64(l1GasFeeWei)))
```

Comment:
Can this be build directly instead of added?

I am not sure how @endulab has been doing it for communities but I believe they call directly:
https://github.com/status-im/status-go/blob/590e35272c8ea3682427a3f3a9df043f15539656/services/wallet/api.go#L396-L399

if this is the case he will need to update this api endpoint but also consider this

Reply:
Indeed, I use this function.

Reply:
> Can this be build directly instead of added?

@alaibe if we have the following params `network, addrTo, addrFrom, token, amountIn` on the client side, we can add a new endpoint and call it directly from the client, if that's what you meant?

Reply:
Sync up with @endulab offline, will create a follow up issue